addons/binding/org.openhab.binding.nest/ESH-INF/thing/bridge.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="nest" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<bridge-type id="bridge">
+		<label>Nest API Source</label>
+		<description>Provides the layer to talk to the nest apis</description>
+		<config-description>
+			<parameter-group name="nest">
+				<advanced>false</advanced>
+				<description>The pieces for talking to nest</description>
+				<label>Nest Connection</label>
+				<context></context>
+			</parameter-group>
+			<parameter-group name="binding">
+				<advanced>false</advanced>
+				<description>Local settings</description>
+				<label>Binding settings</label>
+				<context></context>
+			</parameter-group>
+
+			<parameter name="clientId" type="text" groupName="nest">
+				<label>Client id</label>
+				<description>The client id from the nest product page</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="clientSecret" type="text" groupName="nest">
+				<label>Client secret</label>
+				<description>The client secret from the nest product page</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="pincode" type="text" groupName="nest">
+				<label>Pincode</label>
```

Comment:
No, nest has a way of doing this with a onetime pincode.  You never need to enter your password, and this pincode is swapped for an access token when you connect (and only works once).